### PR TITLE
[FEATURE] Document record objects

### PR DESCRIPTION
**Pull request by linawolf**
Resolves: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/1000
Related: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/918
Releases: main

### CODE REVIEW

**Comment by brotkrueml** [comment]
The related link in the PR description seems to be wrong.

**Comment by linawolf** [comment]
thx, fixed it accidently deleted last digit

File: Documentation/ApiOverview/DatabaseRecords/Index.rst [hunk 1]
```diff
@@ -95,6 +101,22 @@ subtypes in use where the field `CType` contains the value "list" and the field
 `list-type` contains the actual type. This second level of types exists for
 historic reasons. Read more about it in chapter :ref:`content-element-and-plugin`.
 
+.. _database-record-objects:
```
Comment (froemken) [inline Documentation/ApiOverview/DatabaseRecords/Index.rst:104]:
```suggestion
..  _database-record-objects:
```

File: Documentation/ApiOverview/DatabaseRecords/RecordObjects.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
+:navigation-title: Record objects
+.. include:: /Includes.rst.txt
+.. _record_objects:
```
Comment (froemken) [inline Documentation/ApiOverview/DatabaseRecords/RecordObjects.rst:3]:
```suggestion
..  include:: /Includes.rst.txt
..  _record_objects:
```

File: Documentation/ApiOverview/DatabaseRecords/RecordObjects.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+Provide Records in TypoScript
+=============================
+
+In TypoScript you can use the
+:ref:`RecordTransformationProcessor <t3tsref:RecordTransformationProcessor>`,
+usually in combination with the :ref:`DatabaseQueryProcessor <t3tsref:DatabaseQueryProcessor>`
+to pass record objects to the Fluid templating engine.
+
+.. _record_objects_php:
```
Comment (froemken) [inline Documentation/ApiOverview/DatabaseRecords/RecordObjects.rst:33]:
```suggestion
..  _record_objects_php:
```

File: Documentation/ApiOverview/DatabaseRecords/RecordObjects.rst [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+Provide records in PHP
+======================
+
+..  note::
+    Creating record objects in PHP is considered experimental.
+
+In PHP a record object can be created by the
+:php:`\TYPO3\CMS\Core\Domain\RecordFactory`.
+
+.. _record_objects_fluid:
```
Comment (froemken) [inline Documentation/ApiOverview/DatabaseRecords/RecordObjects.rst:44]:
```suggestion
..  _record_objects_fluid:
```

**Review by froemken — APPROVED** [review]
Awesome feature. Yeah.

**Pull request merged**